### ISSUE TITLE
RPC errors lose fields in transfer from agent to host

### ISSUE THREAD

**Issue by dweinstein** [frida/frida#696]
Node error practices seem to have folks use fields on the error object for additional error context. These error fields are currently lost between the agent code and the host via the RPC interface. I was wondering if we might consider ensuring the full error object is transmitted back over the RPC interface. I have a PoC in progress for this 

```diff
Entering 'capstone'
Entering 'frida-clr'
Entering 'frida-core'
Entering 'frida-gum'
diff --git a/bindings/gumjs/runtime/message-dispatcher.js b/bindings/gumjs/runtime/message-dispatcher.js
index ff939c04..57d633bc 100644
--- a/bindings/gumjs/runtime/message-dispatcher.js
+++ b/bindings/gumjs/runtime/message-dispatcher.js
@@ -49,13 +49,13 @@ function MessageDispatcher() {
             reply(id, 'ok', value);
           })
           .catch(error => {
-            reply(id, 'error', error.message, [error.name, error.stack]);
+            reply(id, 'error', error.message, [error.name, error.stack, error]);
           });
         } else {
           reply(id, 'ok', result);
         }
       } catch (e) {
-        reply(id, 'error', e.message, [e.name, e.stack]);
+        reply(id, 'error', e.message, [e.name, e.stack, e]);
       }
     } else if (operation === 'list') {
       reply(id, 'ok', Object.keys(exports));
Entering 'frida-node'
diff --git a/lib/script.ts b/lib/script.ts
index 7bc03de..2b8787b 100644
--- a/lib/script.ts
+++ b/lib/script.ts
@@ -176,10 +176,13 @@ class ScriptServices extends SignalAdapter implements RpcController {
             if (operation === RpcOperation.Ok) {
                 value = (data !== null) ? data : params[0];
             } else {
-                const [message, name, stack] = params;
+                const [message, name, stack, rawErr] = params;
                 error = new Error(message);
                 error.name = name;
                 error.stack = stack;
+                Object.assign(error, rawErr);
             }
 
             callback(error, value);
Entering 'frida-python'
Entering 'frida-qml'
Entering 'frida-swift'
Entering 'frida-tools'
Entering 'releng/meson'
```

**Comment by oleavr** [comment]
Great idea – let's do it!

**Comment by dweinstein** [comment]
This is Closed via - frida/frida-gum#309 frida/frida-node#32
